system/services/notifications/NotificationService.cfc

```diff
@@ -116,30 +116,59 @@ component autodoc=true displayName="Notification Service" {
 	 *
 	 * @userId.hint id of the admin user whose unread notification count we wish to retrieve
 	 */
-	public numeric function getUnreadNotificationCount( required string userId ) autodoc=true {
+	public numeric function getUnreadNotificationCount(
+		  required string userId
+		, required numeric maxRows 
+	) autodoc=true {
 		var queryResult = _getConsumerDao().selectData(
```

Comment:
This should be changed to *not use maxrows* and, instead, just do:

```
return _getConsumerDao().selectData(
      filter = {...} // full filter here, abbreviating for example
    , useCache = false
    , recordCountOnly = true // can pass this to selectData to efficiently get number of records
);
```

This avoids fetching records just to count them.

Reply:
Hi @DominicWatson amended as per request 👌